test/org/sagebionetworks/bridge/play/controllers/ReportControllerTest.java

```diff
@@ -410,8 +415,76 @@ public void adminCanDeleteParticipantIndex() {
+        verify(mockReportService).updateReportIndex(eq(ReportType.STUDY), reportDataIndex.capture());
+        ReportIndex index = reportDataIndex.getValue();
+        assertTrue(index.isPublic());
+        assertEquals(REPORT_ID, index.getIdentifier());
+        assertEquals("api:STUDY", index.getKey());
+        
+        TestUtils.assertResult(result, 200, "Report index updated.");
+    }
+    
+    @Test
+    public void canUpdateParticipantReportIndex() throws Exception {
+        TestUtils.mockPlayContextWithJson("{\"public\":true}");
+        
+        Result result = controller.updateParticipantReportIndex(REPORT_ID);
+        
+        verify(mockReportService).updateReportIndex(eq(ReportType.PARTICIPANT), reportDataIndex.capture());
+        ReportIndex index = reportDataIndex.getValue();
+        assertTrue(index.isPublic());
+        assertEquals(REPORT_ID, index.getIdentifier());
+        assertEquals("api:PARTICIPANT", index.getKey());
+        
+        TestUtils.assertResult(result, 200, "Report index updated.");
+    }
+    
+    private static final TypeReference<DateRangeResourceList<? extends ReportData>> REPORT_REF = new TypeReference<DateRangeResourceList<? extends ReportData>>() {
+    };
+    
+    @Test
+    public void canGetPublicStudyReport() throws Exception {
+        ReportIndex index = ReportIndex.create();
+        index.setPublic(true);
+        index.setIdentifier(REPORT_ID);
+        ReportTypeResourceList<? extends ReportIndex> list = new ReportTypeResourceList<>(
+                Lists.newArrayList(index), ReportType.STUDY);
+        doReturn(list).when(mockReportService).getReportIndices(TEST_STUDY, ReportType.STUDY);
+        
+        doReturn(makeResults(START_DATE, END_DATE)).when(mockReportService).getStudyReport(session.getStudyIdentifier(),
+                REPORT_ID, START_DATE, END_DATE);
+        
+        Result result = controller.getPublicStudyReport(
+                TEST_STUDY.getIdentifier(), REPORT_ID, START_DATE.toString(), END_DATE.toString());
+        
+        assertEquals(200, result.status());
+        DateRangeResourceList<? extends ReportData> reportData = BridgeObjectMapper.get()
+                .readValue(Helpers.contentAsString(result), REPORT_REF);
+        assertEquals(2, reportData.getItems().size());
+        
+        verify(mockReportService).getReportIndices(TEST_STUDY, ReportType.STUDY);
+        verify(mockReportService).getStudyReport(TEST_STUDY, REPORT_ID, START_DATE, END_DATE);
+    }
+    
+    @Test(expected = EntityNotFoundException.class)
+    public void missingPublicStudyReturns404() throws Exception {
+        controller.getPublicStudyReport(TEST_STUDY.getIdentifier(), "does-not-exist", "2016-05-02", "2016-05-09");
+    }
+    
+    @Test(expected = EntityNotFoundException.class)
+    public void privatePublicStudyReturns404() throws Exception {
+        controller.getPublicStudyReport(TEST_STUDY.getIdentifier(), REPORT_ID, "2016-05-02", "2016-05-09");
     }
```

Comment:
I don't see anywhere in the test where we create this report. Are we sure this test adequately covers the "private report" case?